coap-core/src/main/java/com/mbed/coap/server/block/BlockRequestId.java

```diff
@@ -24,7 +25,11 @@ class BlockRequestId {
     private final String uriPath;
     private final InetSocketAddress sourceAddress;
 
-    public BlockRequestId(String uriPath, InetSocketAddress sourceAddress) {
+    static BlockRequestId from(CoapRequest request) {
+        return new BlockRequestId(request.options().getUriPath(), request.getPeerAddress());
+    }
```

Comment:
If you wanted to support concurrent block transfer you could add request-tag as part of the id right ?

Reply:
Exactly